promgen/views.py

```diff
@@ -1374,3 +1376,31 @@ def post(self, request, pk):
         return JsonResponse(
             {request.POST["target"]: render_to_string("promgen/ajax_clause_check.html", result)}
         )
+
+
+class PromqlQuery(View):
```

Comment:
Some of these views we've put in `proxy.py` (since we're proxying over the Prometheus APIs) I'm also wondering that if we add http basic auth support here, then we need to make the rest of our proxy routes aware of the new basic auth settings

Reply:
True, all of them should check if authorization is needed.